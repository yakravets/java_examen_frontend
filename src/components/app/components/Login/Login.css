.login-form {
    border: 1px #1890ff solid;
    border-radius: 10px;
    padding: 10px;
    justify-content: center;
}