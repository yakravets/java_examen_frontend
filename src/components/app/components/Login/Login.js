import React from 'react';
import { Form, Input, Button, Checkbox, Divider } from 'antd';
import './Login.css';

export default function Login(){
  return (
      <>
        <div className='login-form'>
            <Form name="basic" labelCol={{span: 8,}}
                wrapperCol={{span: 8,}}
                initialValues={{
                    remember: true,
                }}
                onFinish={onFinish}
                onFinishFailed={onFinishFailed}
                >
                <Divider />
                <Form.Item label="Username" name="username"
                    rules={[
                    {
                        required: true,
                        message: 'Please input your username!',
                    },
                    ]}
                >
                    <Input />
                </Form.Item>
                <Form.Item
                    label="Password"
                    name="password"
                    rules={[
                    {
                        required: true,
                        message: 'Please input your password!',
                    },
                    ]}
                >
                    <Input.Password />
                </Form.Item>
                <Form.Item name="remember" valuePropName="checked"
                    wrapperCol={{
                        offset: 8,
                        span: 16,
                    }}
                >
                    <Checkbox>Remember me</Checkbox>
                </Form.Item>
                <Form.Item
                    wrapperCol={{
                        offset: 8,
                        span: 16,
                    }}
                >
                    <Button type="primary" htmlType="submit">
                        Submit
                    </Button>
                </Form.Item>
            </Form>
        </div>
    </>
  );
}

const onFinish = (values) => {
    console.log('Success:', values);
  };

const onFinishFailed = (errorInfo) => {
    console.log('Failed:', errorInfo);
  };