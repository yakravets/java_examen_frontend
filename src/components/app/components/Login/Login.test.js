import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";

import Login from './Login';

it("renders correctly", () => {
    const tree = shallow(<Login />);
    expect(toJson(tree)).toMatchSnapshot();
});

it("renders without crashing", () => {
    shallow(<Login />);
});