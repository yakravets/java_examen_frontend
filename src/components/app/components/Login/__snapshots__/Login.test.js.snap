// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <div
    className="login-form"
  >
    <ForwardRef(InternalForm)
      initialValues={
        Object {
          "remember": true,
        }
      }
      labelCol={
        Object {
          "span": 8,
        }
      }
      name="basic"
      onFinish={[Function]}
      onFinishFailed={[Function]}
      wrapperCol={
        Object {
          "span": 8,
        }
      }
    >
      <Divider />
      <FormItem
        label="Email"
        name="login"
        rules={
          Array [
            Object {
              "message": "Please input your email!",
              "required": true,
            },
          ]
        }
      >
        <Input
          type="text"
        />
      </FormItem>
      <FormItem
        label="Password"
        name="password"
        rules={
          Array [
            Object {
              "message": "Please input your password!",
              "required": true,
            },
          ]
        }
      >
        <Password
          action="click"
          iconRender={[Function]}
          visibilityToggle={true}
        />
      </FormItem>
      <FormItem
        name="remember"
        valuePropName="checked"
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Checkbox>
          Remember me
        </Checkbox>
      </FormItem>
      <FormItem
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Button
          htmlType="submit"
          type="primary"
        >
          Submit
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
  </div>
</Fragment>
`;
